### ISSUE TITLE
Create a DI adapter for Microsoft.Extensions.DependencyInjection

### ISSUE THREAD

**Issue by alexvaluyskiy** [AkkaNetContrib/Home#7]
https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection

**Comment by modernist** [comment]
I'm not sure if it's feasible to create an adapter for Microsoft.Extensions.DependencyInjection since it does not support child containers: https://github.com/aspnet/DependencyInjection/issues/246

**Comment by tintoy** [comment]
BTW, just implemented one of these - for the simple case at least, it's pretty easy:

https://gist.github.com/tintoy/37ab414f8fd625258fc82d4094ff6dd3

**Comment by MikeBairdRocks** [comment]
I say put a repo out there using @tintoy 's gist. I for one am using `Microsoft.Extensions.DependencyInjection` as my default for all .Net core projects now and it would be nice to see Akka.net support it.

I'll be willing to get the repo going if it would help.

**Comment by MikeBairdRocks** [comment]
Here is the repo: https://github.com/cecilphillip/Akka.DI.Microsoft
I'm not sure what's next but build works and all tests pass.

**Comment by mika76** [comment]
Hey @MikeBairdRocks any reason that repo was archived?

**Comment by MikeBairdRocks** [comment]
> Hey @MikeBairdRocks any reason that repo was archived?

It died with this thread. I didn't have the time to maintain it. @cecilphillip has taken it over but has not done anything with it yet.

**Comment by mika76** [comment]
Thanks @MikeBairdRocks I'll definitely do that.

@Aaronontheweb I wonder - would this not be a good repo to assimilate into AkkaNetContrib?

**Comment by cecilphillip** [comment]
Yes, @MikeBairdRocks is correct. I haven't had much time to put into it either to be honest.

**Comment by mika76** [comment]
@cecilphillip Is it stable as is? What I mean is, is it pretty much considered done and working? Have you used it in prod at all?
